web/src/gear_menu.ts

```diff
@@ -194,17 +195,26 @@ export function initialize(): void {
                     );
                 information_density.update_information_density_settings($(this), changed_property);
                 information_density.enable_or_disable_control_buttons($popper);
-                e.preventDefault();
-            });
 
-            const resizeObserver = new ResizeObserver(() => {
-                requestAnimationFrame(() => {
-                    void instance.popperInstance?.update();
-                });
+                if (changed_property === "web_font_size_px") {
+                    // We do not want to display the arrow once font size is
+                    // changed because popover will be detached from the gear
+                    // icon as we do not change the font size in popover.
+                    $("#gear-menu-dropdown").closest(".tippy-box").find(".tippy-arrow").hide();
+                }
+
+                e.preventDefault();
             });
-            resizeObserver.observe(document.querySelector("#gear-menu-dropdown")!);
 
             information_density.enable_or_disable_control_buttons($popper);
+
+            // We do not want font size of the popover to change when changing
+            // font size using the buttons in popover, so that the buttons do
+            // not shift.
+            const font_size = 1.0714 * user_settings.web_font_size_px;
```

Comment:
It would be good to document here, and possibly declare a constant, so we know that `1.0714` comes from 15px at 14px/1em.